test/jb_eth_payment_terminal/use_allowance_of.test.js

```diff
@@ -0,0 +1,443 @@
+      );
+
+    // Terminal should be out of ETH
+    expect(await ethers.provider.getBalance(jbEthPaymentTerminal.address)).to.equal(0);
+
+    // Beneficiary should have a larger balance
+    expect(await ethers.provider.getBalance(beneficiary.address)).to.equal(
+      initialBeneficiaryBalance.add(AMOUNT_MINUS_FEES),
+    );
+  });
+
+  it('Should send funds from overflow, with fees held, then process fees, and emit event', async function () {
+    const {
+      beneficiary,
+      CURRENCY_ETH,
+      ETH_ADDRESS,
+      jbEthPaymentTerminal,
+      mockJbDirectory,
+      mockJbEthPaymentTerminalStore,
+      projectOwner,
+      terminalOwner,
+      timestamp,
+    } = await setup();
+
+    const newFundingCycle = {
+      number: FUNDING_CYCLE_NUM,
+      configuration: timestamp,
+      basedOn: timestamp,
+      start: timestamp,
+      duration: 0,
+      weight: WEIGHT,
+      discountRate: 0,
+      ballot: ethers.constants.AddressZero,
+      metadata: packFundingCycleMetadata({ holdFees: 1 }), // Hold fees
+    };
+
+    await mockJbEthPaymentTerminalStore.mock.recordUsedAllowanceOf
+      .withArgs(PROJECT_ID, /* amount */ AMOUNT, CURRENCY_ETH, /* minReturnedWei */ AMOUNT)
+      .returns(newFundingCycle, AMOUNT);
+
+    await mockJbEthPaymentTerminalStore.mock.recordPaymentFrom
+      .withArgs(
+        projectOwner.address,
+        AMOUNT - AMOUNT_MINUS_FEES,
+        JUICEBOX_PROJECT_ID,
+        ethers.BigNumber.from(0).or(ethers.BigNumber.from(projectOwner.address).shl(1)),
+        /* minReturnedTokens */ 0,
+        /* memo */ 'Fee from @' + ethers.utils.parseBytes32String(HANDLE) + PADDING,
+        /* delegateMetadata */ '0x',
+      )
+      .returns(newFundingCycle, WEIGHT, AMOUNT, MEMO);
+
+    await mockJbDirectory.mock.primaryTerminalOf
+      .withArgs(1, ETH_ADDRESS)
+      .returns(jbEthPaymentTerminal.address);
+
+    // Give terminal sufficient ETH
+    await ethers.provider.send('hardhat_setBalance', [
+      jbEthPaymentTerminal.address,
+      '0x' + AMOUNT_MINUS_FEES.toString(16),
+    ]);
+
+    // Set fee to default 5%
+    await jbEthPaymentTerminal.connect(terminalOwner).setFee(DEFAULT_FEE);
+
+    // Use allowance and hold fee
+    await jbEthPaymentTerminal
+      .connect(projectOwner)
+      .useAllowanceOf(
+        PROJECT_ID,
+        AMOUNT,
+        CURRENCY_ETH,
+        /* minReturnedWei */ AMOUNT,
+        beneficiary.address,
+      );
+
+    // Should be holding fees in the contract
+    expect(await jbEthPaymentTerminal.heldFeesOf(PROJECT_ID)).to.eql([
+      [
+        ethers.BigNumber.from(AMOUNT),
+        DEFAULT_FEE,
+        projectOwner.address,
+        'Fee from @' + ethers.utils.parseBytes32String(HANDLE) + PADDING,
+      ],
+    ]);
+
+    // Process held fees
+    const tx = await jbEthPaymentTerminal.connect(projectOwner).processFees(PROJECT_ID);
+
+    expect(await tx).to.emit(jbEthPaymentTerminal, 'ProcessFees');
+    // .withArgs(
```

Comment:
why's this commented out? can it be removed or does it add context?

Reply:
Added comment for context